src/interface/src/app/validators/passwords.ts

```diff
@@ -0,0 +1,86 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+// the possible custom errors
+interface PasswordFieldsErrors {
+  newPasswordMustBeNew: boolean;
+  newPasswordsMustMatch: boolean;
+  mustContainNumber: boolean;
```

Comment:
I'm sorry to say that I was told we weren't doing the number, upper, or lower requirements now, just enforcing length, as a password that's long enough still presumably has enough entropy to be secure. However, I've also been awaiting the California security requirements to confirm this, since we ultimately would be audited by them.

Reply:
@lastminutediorama thanks for clarifying this! Is the minimum length still 8 ?